mitiq/mitiq_pyquil/tests/test_pyquil_compiler.py

```diff
@@ -0,0 +1,319 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""NB: Copied in large part from rigetti/forest-benchmarking (Apache-2.0)
+and modified to test a larger gateset.
+"""
+
+import inspect
+import random
+import itertools
+from math import pi
+
+import numpy as np
+import pytest
+
+from pyquil.gates import (
+    CCNOT,
+    CNOT,
+    CPHASE,
+    CSWAP,
+    CZ,
+    H,
+    I,
+    ISWAP,
+    PHASE,
+    RX,
+    RY,
+    RZ,
+    S,
+    SWAP,
+    T,
+    X,
+    XY,
+    Y,
+    Z,
+)
+from pyquil.quil import Program
+
+from mitiq.mitiq_pyquil.compiler import (
+    basic_compile,
+    _CCNOT,
+    _CNOT,
+    _CPHASE,
+    _H,
+    _ISWAP,
+    _RY,
+    _S,
+    _SWAP,
+    _T,
+    _X,
+    _Y,
+    _Z,
+    match_global_phase,
+)
+
+try:
+    from pyquil.simulation.tools import program_unitary
+
+    unitary_tools = True
+except ImportError:
+    unitary_tools = False
```

Comment:
Why is this needed? Is this to support different pyquil versions?

Reply:
Yeah, and it was copied over from forest-benchmarking -- I'll go ahead and delete it, because we require a later version of pyquil anyways.